File: GanttChartLightLibraryDemos/Demos/Samples.Resources/WPF-CSharp/GanttChartDataGrid/BarTemplating/Interruption.cs
using System;
using System.ComponentModel;
using DlhSoft.Windows.Controls;

namespace Demos.WPF.CSharp.GanttChartDataGrid.BarTemplating
{
    class Interruption : INotifyPropertyChanged
    {
        public CustomGanttChartItem Item { get; internal set; }
        public IGanttChartView GanttChartView { get { return Item != null ? Item.GanttChartView : null; } }

        private DateTime start;
        public DateTime Start
        {
            get { return start; }
            set
            {
                start = value;
                OnPropertyChanged("Start");
            }
        }
        public double ComputedLeft
        {
            get { return GanttChartView.GetPosition(Start) - GanttChartView.GetPosition(Item.Start); }
        }

        private DateTime finish;
        public DateTime Finish
        {
            get { return finish; }
            set
            {
                finish = value;
                OnPropertyChanged("Finish");
            }
        }
        public double ComputedWidth
        {
            get { return GanttChartView.GetPosition(Finish) - GanttChartView.GetPosition(Item.Start) - ComputedLeft; }
        }

        protected internal virtual void OnPropertyChanged(string propertyName)
        {
            if (PropertyChanged != null)
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            switch (propertyName)
            {
                case "Start":
                    OnPropertyChanged("ComputedLeft");
                    break;
            }
            switch (propertyName)
            {
                case "Start":
                case "Finish":
                    OnPropertyChanged("ComputedWidth");
                    break;
            }
        }
        public event PropertyChangedEventHandler PropertyChanged;
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples.Resources/WPF-VisualBasic/GanttChartDataGrid/BarTemplating/Interruption.vb
Imports System.ComponentModel
Imports DlhSoft.Windows.Controls

Friend Class Interruption
    Implements INotifyPropertyChanged

    Private privateItem As CustomGanttChartItem
    Public Property Item() As CustomGanttChartItem
        Get
            Return privateItem
        End Get
        Friend Set(ByVal value As CustomGanttChartItem)
            privateItem = value
        End Set
    End Property
    Public ReadOnly Property GanttChartView() As IGanttChartView
        Get
            Return If(Item IsNot Nothing, Item.GanttChartView, Nothing)
        End Get
    End Property

    Private startValue As Date
    Public Property Start() As Date
        Get
            Return startValue
        End Get
        Set(value As Date)
            startValue = value
            OnPropertyChanged("Start")
        End Set
    End Property
    Public ReadOnly Property ComputedLeft() As Double
        Get
            Return GanttChartView.GetPosition(Start) - GanttChartView.GetPosition(Item.Start)
        End Get
    End Property

    Private finishValue As Date
    Public Property Finish() As Date
        Get
            Return finishValue
        End Get
        Set(value As Date)
            finishValue = value
            OnPropertyChanged("Finish")
        End Set
    End Property
    Public ReadOnly Property ComputedWidth() As Double
        Get
            Return GanttChartView.GetPosition(Finish) - GanttChartView.GetPosition(Item.Start) - ComputedLeft
        End Get
    End Property

    Protected Friend Overridable Sub OnPropertyChanged(propertyName As String)
        RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        Select Case propertyName
            Case "Start"
                OnPropertyChanged("ComputedLeft")
        End Select
        Select Case propertyName
            Case "Start", "Finish"
                OnPropertyChanged("ComputedWidth")
        End Select
    End Sub
    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
End Class
